packages/grafana-data/src/dataframe/StreamingDataFrame.ts

```diff
@@ -163,9 +197,18 @@ export class StreamingDataFrame implements DataFrame {
         });
       }
 
-      let curValues = this.fields.map((f) => f.values.buffer);
+      let appended = values;
+      this.packetInfo.length = values[0].length;
 
-      let appended = circPush(curValues, values, this.options.maxLength, this.timeFieldIndex, this.options.maxDelta);
+      if (this.alwaysReplace || !this.length) {
```

Comment:
Nit: if the DF is empty, shouldn't the action be Append? I guess this is cosmetics.